cypress/integration/plugins/security-dashboards-plugin/tenancy_change_on_shortlink.js

```diff
@@ -0,0 +1,139 @@
+/*
+
+* Copyright OpenSearch Contributors
+
+* SPDX-License-Identifier: Apache-2.0
+
+*/
+import { CURRENT_TENANT } from '../../../utils/commands';
+import tenantDescription from '../../../fixtures/plugins/security-dashboards-plugin/tenants/testTenant.json';
+
+const tenantName = 'test';
+
+if (Cypress.env('SECURITY_ENABLED')) {
+  describe('Multi Tenancy Tests: ', () => {
+    before(() => {
+      cy.server();
+      cy.createTenant(tenantName, tenantDescription);
+
+      try {
+        cy.createIndexPattern(
+          'index-pattern1',
+          {
+            title: 's*',
+            timeFieldName: 'timestamp',
+          },
+          indexPatternGlobalTenantHeaderSetUp
+        );
+      } catch (error) {
+        // Ignore
+      }
+      try {
+        cy.createIndexPattern(
+          'index-pattern2',
+          {
+            title: 'se*',
+            timeFieldName: 'timestamp',
+          },
+          indexPatternPrivateTenantHeaderSetUp
+        );
+      } catch (error) {
+        // Ignore
+      }
+    });
+
+    it('Tests that when the short URL is copied and pasted, it will route correctly with the right tenant', function () {
+      cy.visit('/app/dashboards#create', {
+        excludeTenant: true,
+        onBeforeLoad(win) {
+          // set up session storage as we would expect to emulate browser
+          win.sessionStorage.setItem(
+            'opendistro::security::tenant::show_popup',
+            false
+          );
+          win.localStorage.setItem(
+            'opendistro::security::tenant::saved',
+            '__user__'
+          );
+        },
+      });
+
+      // 1. Create a dashboard (to be able to share it later)
+      cy.getElementByTestId('dashboardSaveMenuItem').click();
+
+      const randomNumber = Cypress._.random(0, 1e6);
+      const dashboardName = 'Cypress test: My dashboard - ' + randomNumber;
+      cy.getElementByTestId('savedObjectTitle').type(dashboardName);
+
+      cy.intercept({
+        method: 'POST',
+        url: '/api/saved_objects/_bulk_get',
+      }).as('waitForReloadingDashboard');
+      cy.getElementByTestId('confirmSaveSavedObjectButton').click();
+      cy.wait('@waitForReloadingDashboard');
+
+      // 2. Open top share navigation to access copy short url
+      cy.get('[data-test-subj="shareTopNavButton"]').click();
+      cy.getElementByTestId('sharePanel-Permalinks').click();
+
+      // 3. Create the short url, wait for response
+      cy.intercept('POST', '/api/shorten_url').as('getShortUrl');
+      // If the url already contains the tenant parameter, it will be stored in the short url. That will work in the app
+      // but would render this test useless. We're testing that resolved short urls without the tenant parameter work as well.
+      cy.url().should('not.contain', 'security_tenant');
+      cy.getElementByTestId('createShortUrl').click();
+      cy.wait('@getShortUrl');
+
+      function switchTenantTo(newTenant) {
```

Comment:
Looks like this function is repeated here? Can this be extracted to a common place?